src/Deserialization/Transaction.purs

```diff
@@ -0,0 +1,642 @@
+module Deserialization.Transaction where
+
+import Prelude
+
+import Contract.Address
+  ( RequiredSigner(RequiredSigner)
+  , Slot(Slot)
+  , StakeCredential
+  )
+import Contract.Numeric.Rational (reduce)
+import Contract.Prim.ByteArray (ByteArray)
+import Contract.Scripts (Ed25519KeyHash)
+import Contract.Transaction
+  ( AuxiliaryData(AuxiliaryData)
+  , AuxiliaryDataHash
+  , Certificate(StakeDeregistration, StakeRegistration, StakeDelegation)
+  , CostModel(CostModel)
+  , Costmdls(Costmdls)
+  , Epoch(Epoch)
+  , ExUnitPrices
+  , ExUnits
+  , GeneralTransactionMetadata
+  , GenesisHash
+  , Language(PlutusV1)
+  , Mint(Mint)
+  , Nonce(HashNonce, IdentityNonce)
+  , ProposedProtocolParameterUpdates(ProposedProtocolParameterUpdates)
+  , ProtocolParamUpdate
+  , ProtocolVersion
+  , ScriptDataHash
+  , TransactionMetadatum(MetadataList, MetadataMap, Bytes, Int, Text)
+  , TransactionMetadatumLabel(TransactionMetadatumLabel)
+  , TxBody(TxBody)
+  , Update
+  )
+import Contract.Value (Coin(Coin), NonAdaAsset(NonAdaAsset), TokenName)
+import Control.Lazy (fix)
+import Data.Bifunctor (bimap, lmap)
+import Data.BigInt (BigInt)
+import Data.BigInt as BigInt
+import Data.Bitraversable (bitraverse)
+import Data.Map as M
+import Data.Maybe (Maybe)
+import Data.Newtype (wrap)
+import Data.Ratio (Ratio)
+import Data.Traversable (traverse, for)
+import Data.Tuple (Tuple)
+import Data.UInt (UInt)
+import Data.UInt as UInt
+import Data.Variant (inj)
+import Deserialization.BigNum (bigNumToBigInt')
+import Deserialization.Error
+  ( Err
+  , FromCslRepError
+  , _fromCslRepError
+  , addErrTrace
+  , cslErr
+  , fromCslRepError
+  )
+import Deserialization.FromBytes (fromBytes')
+import Deserialization.UnspentOutput (convertInput, convertOutput)
+import Deserialization.WitnessSet
+  ( convertNativeScripts
+  , convertPlutusScripts
+  , convertWitnessSet
+  )
+import Error (E, notImplementedError)
+import FfiHelpers
+  ( ContainerHelper
+  , ErrorFfiHelper
+  , MaybeFfiHelper
+  , containerHelper
+  , errorHelper
+  , maybeFfiHelper
+  )
+import Serialization.Address (RewardAddress, intToNetworkId)
+import Serialization.Hash (ScriptHash)
+import Serialization.Types (NativeScripts, PlutusScripts)
+import Serialization.Types as Csl
+import Type.Row (type (+))
+import Types.Transaction as T
+import Types.Value (scriptHashAsCurrencySymbol, tokenNameFromAssetName)
+
+-- | Deserializes CBOR encoded transaction to a CTL's native type.
+deserializeTransaction
+  :: forall (r :: Row Type). { txCbor :: ByteArray } -> Err r T.Transaction
```

Comment:
Why not just `ByteArray`?

Reply:
To make Cbor keyword mandatory, as soon as we have newtype wrappers for Cbor Bytearrays we can switch to direct types

Reply:
Could you add a note with the issue URL please?